File: application/server/src/test/kotlin/io/github/siyual_park/application/server/dummy/DummyCreateClientRequest.kt
package io.github.siyual_park.application.server.dummy

import io.github.siyual_park.application.server.dto.request.CreateClientRequest
import io.github.siyual_park.client.entity.ClientType
import io.github.siyual_park.test.DummyNameFactory
import io.github.siyual_park.test.resolveNotNull
import java.net.URL
import java.util.Optional

object DummyCreateClientRequest {
    data class Template(
        val name: Optional<String>? = null,
        val type: Optional<ClientType>? = null,
        val origin: Optional<URL>? = null
    )

    fun create(template: Template? = null): CreateClientRequest {
        return CreateClientRequest(
            name = resolveNotNull(template?.name) { DummyNameFactory.create(10) },
            type = resolveNotNull(template?.type) { ClientType.PUBLIC },
            origin = resolveNotNull(template?.origin) { URL("http://localhost:8080") },
        )
    }
}

File: k6-tests/src/dummy/dummy-create-client-request.ts
// @ts-ignore
import { randomString } from 'https://jslib.k6.io/k6-utils/1.2.0/index.js';

import { CreateClientRequest } from '../request';

function dummyCreateClientRequest(): CreateClientRequest {
  return {
    name: randomString(10),
    type: 'confidential',
    origin: 'http://localhost:8080',
  };
}

export default dummyCreateClientRequest;
